src/libDirectoryService/DirectoryService.h

```diff
@@ -73,10 +73,11 @@ class DirectoryService : public Executable, public Broadcastable
     std::condition_variable cv_scheduleFinalBlockConsensus;
 
     // PoW common variables
-    std::mutex m_mutexAllPoWs;
-    std::map<PubKey, Peer> m_allPoWConns;
     std::mutex m_mutexAllPoWConns;
+    std::map<PubKey, Peer> m_allPoWConns;
     std::vector<unsigned char> m_PoWConsensusMessage;
+    std::mutex m_mutexAllPoWCounter;
+    std::map<PubKey, uint8_t> m_AllPoWCounter;
```

Comment:
What does this m_AllPoWCounter use for?

Reply:
number of PoW submissions. As there is 2 level of difficult, a node may submit twice or more. The counter is there to track the number of submission and prevent spamming of submission